lldp_dcbx.c

```diff
@@ -431,6 +430,8 @@ struct lldp_module * dcbx_register(void)
 	LLDPAD_DBG("%s: dcbx register done\n", __func__);
 	return mod;
 out_err:
+	free(mod);
+	free(dud);
```

Comment:
You need to initialize `mod` and `dud` to `NULL` in the function prologue as they may be uninitialized under some error code paths that reach here leading to a crash instead of a leak.